runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java

```diff
@@ -677,6 +680,40 @@ private void processInventoryLoot(String event, LootRecordType lootRecordType, I
 		}
 	}
 
+	private boolean processHerbSackLoot(int timestamp, String event, LootRecordType lootRecordType)
+	{
+		List<ItemStack> herbs = new ArrayList<>();
+		for (Object messageObject : client.getMessages())
+		{
+			MessageNode messageNode = (MessageNode) messageObject;
+			if (messageNode.getTimestamp() == timestamp && messageNode.getType() == ChatMessageType.SPAM)
+			{
+				String herbName;
```

Comment:
As I understand from the code below, there are two possible chat messages which can appear:

* You put the Grimy Guam leaf herb into your herb sack.
* Your herb sack is too full to hold the Grimy Guam leaf herb.

Wouldn't a pattern matching `.+(Grimy .+) herb.+` match the herb name fully without any of the extra logic below? Also, the pattern matching/replacing should be replaced with a compiled `Pattern` object and a matcher on the messageNode string.

Reply:
There is also another message when you manually add a herb to your sack which is:

- You add the Grimy Guam leaf to your sack.

But I don't think its possible to manually add herbs in the same timestamp as harvesting Herbiboar.

Reply:
Pattern ended up being `.+(Grimy .+?) herb.+`